bindata/network/ovn-kubernetes/006-ovs-node.yaml

```diff
@@ -42,26 +42,49 @@ spec:
         - |
           #!/bin/bash
           set -euo pipefail
+          set -x
           if [[ -f "/env/${K8S_NODE}" ]]; then
             set -o allexport
             source "/env/${K8S_NODE}"
             set +o allexport
           fi
           chown -R openvswitch:openvswitch /run/openvswitch
           chown -R openvswitch:openvswitch /etc/openvswitch
+          mkdir -p /run/bin
           function quit {
               # Don't allow ovs-vswitchd to clear datapath flows on exit
               kill -9 $(cat /var/run/openvswitch/ovs-vswitchd.pid 2>/dev/null) 2>/dev/null || true
               kill $(cat /var/run/openvswitch/ovsdb-server.pid 2>/dev/null) 2>/dev/null || true
               exit 0
           }
           trap quit SIGTERM
-          /usr/share/openvswitch/scripts/ovs-ctl start --ovs-user=openvswitch:openvswitch --system-id=random
-          ovs-appctl vlog/set "file:${OVS_LOG_LEVEL}"
-          /usr/share/openvswitch/scripts/ovs-ctl --protocol=udp --dport=6081 enable-protocol
+          #cat > /run/bin/openshift-cno-ovs <<EOF
+          cat <<'EOF' > /run/bin/openshift-cno-ovs
+          #!/bin/bash
+          set -euo pipefail
+          set -x
+          export SYSTEMD_IGNORE_CHROOT=yes
+          echo "systemd ignore chroot set to " \$SYSTEMD_IGNORE_CHROOT
+          systemctl start ovsdb-server
```

Comment:
I don't think this is the correct approach - why wouldn't this operator add machine config injection to start this automatically?

I would expect CNO to contribute daemonsets AND an MCO config injector to all pools (except maybe those are invoking) that depends on OVS

Reply:
There are definitely advantages to that, but a downside is that in order to avoid race conditions, the logic for rendering this config would need to be part of the bootstrap MCO phase at least I'd think.

Otherwise we'd have a situation where the MCO starts and renders a config *without* the CNO setup, then the CNO starts and injects a MC which forces another rollout.

It also gets weird to inject nontrivial *code* via systemd units...the MCO has some of the "kubernetes native shell script" stuff that's a template to run a static pod that runs shell, which got really unwieldly and ended up being moved into a separate operator.

Reply:
Yes, they need to be part of bootstrap MCO and CNO needs to contribute to bootstrap MCO

Reply:
Here's my top level expectation: "anyone can contribute an MC to the cluster that enables a systemd unit to start at boot of the masters trivially"

Reply:
> Here's my top level expectation: "anyone can contribute an MC to the cluster that enables a systemd unit to start at boot of the masters trivially"

And then that operator owns upgrades to that MC right?

Yes, there are definitely advantages to this - actually if we took this approach we could probably move some of the "platform specific" stuff out of the MCO into separate operators, like baremetal-operator could own the `baremetal` platform MC stuff right now.

It'd probably be sufficient to extend `bootkube.sh` to have a static list of operators it walks over and asks them to dump machineconfig files into, then the bootstrap MCS just merges those in when it starts.

Reply:
> It'd probably be sufficient to extend bootkube.sh to have a static list of operators it walks over and asks them to dump machineconfig files into, then the bootstrap MCS just merges those in when it starts. Then they also get injected into the cluster.

💯